plutus-contract/src/Language/Plutus/Contract/StateMachine.hs

```diff
@@ -149,15 +150,16 @@ getOnChainState StateMachineClient{scInstance, scChooser} = mapError (review _SM
     either (throwing _SMContractError) (\s -> pure (s, utxo)) (scChooser states)
 
 -- | Wait until the on-chain state of the state machine instance has changed,
---   and return the new state.
+--   and return the new state, or return 'Nothing' if the instance has been
+--   terminated.
```

Comment:
What happens if the instance hasn't started? Maybe we should return a more specific type than `Maybe` (if there are in fact more than two possibilities?

Reply:
`waitForUpdate` calls `nextTransactionAt`, which returns the next transaction that changes the SM instance address (change = add or remove an output). If the set of UTXOs at the script address is empty after `nextTransactionAt`, then the transaction must have spent one or more inputs from the address and produced zero outputs. So if you call `waitForUpdate` before the instance has started, it will wait until the instance starts and then return the new state.

I'll put that in a comment as well.

(of course there is the underlying issue that we can't tell two instances with the same address apart until we have [thread tokens](https://jira.iohk.io/browse/SCP-119))